src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -71,11 +70,12 @@ private static bool TryDecodeSuppressMessageAttributeData (CustomAttribute attri
 			// and category information can be obtained from warnings themselves.
 			// We only support warnings with code pattern IL####.
 			if (!(attribute.ConstructorArguments[1].Value is string warningId) ||
-				!Regex.IsMatch (warningId, "^IL\\d{4}")) {
+				warningId.Length < 6 ||
+				!warningId.StartsWith ("IL") ||
+				!int.TryParse (warningId.Substring (2, 4), out info.Id)) {
```

Comment:
This means that "IL1234HiThere" will be read as 1234. Probably not what we want. Just check that Length != 6.

Reply:
`SuppressMessageAttributes` allows the user to write anything after a colon, mainly for documentation purposes (e.g., `"IL1234:Suppress this"`). I think we should keep this capability.